packages/patternfly-4/react-core/src/components/Checkbox/Checkbox.tsx

```diff
@@ -77,6 +79,7 @@ export class Checkbox extends React.Component<CheckboxProps> {
           aria-invalid={!isValid}
           aria-label={ariaLabel}
           disabled={isDisabled}
+          ref={elem => elem && (elem.indeterminate = isChecked === null)}
```

Comment:
What's this ref needed for?

Reply:
This prop is can't be set trough react prop for some reason [1]. However plain js api allows to set it trough element, so this ref is to access checkox element. It's probably because indeterminate is a flag and not real prop though.

[1] https://github.com/facebook/react/issues/1798